src/fides/api/ops/models/privacy_experience.py

```diff
@@ -165,26 +297,302 @@ def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperience:
+    ) -> Tuple[Optional[PrivacyExperience], Optional[PrivacyExperience]]:
+        """Load both the overlay and privacy center experience for a given region"""
+        overlay_experience: Optional[PrivacyExperience] = (
+            db.query(PrivacyExperience)
+            .filter(
+                PrivacyExperience.region == region,
+                PrivacyExperience.component == ComponentType.overlay,
+            )
+            .first()
+        )
+
+        privacy_center_experience: Optional[PrivacyExperience] = (
+            db.query(PrivacyExperience)
+            .filter(
+                PrivacyExperience.region == region,
+                PrivacyExperience.component == ComponentType.privacy_center,
+            )
+            .first()
+        )
+
+        return overlay_experience, privacy_center_experience
+
+    def unlink_privacy_experience_config(self, db: Session) -> PrivacyExperience:
+        """Remove config from experience"""
+        return self.update(
+            db,
+            data={
+                "experience_config_id": None,
+                "experience_config_history_id": None,
+            },
+        )
 
 
 class PrivacyExperienceHistory(PrivacyExperienceBase, Base):
-    """Stores historical records of Privacy Experiences for Consent Reporting"""
+    """Stores the history of a privacy experience for a given region for Consent Reporting"""
 
     version = Column(Float, nullable=False, default=1.0)
-    privacy_experience_template_id = Column(
-        String, ForeignKey(PrivacyExperienceTemplate.id_field_path), nullable=True
-    )
     privacy_experience_id = Column(
-        String, ForeignKey(PrivacyExperience.id_field_path), nullable=False
+        String, ForeignKey(PrivacyExperience.id_field_path), nullable=False, index=True
     )
+    experience_config_id = Column(
+        String,
+        ForeignKey(PrivacyExperienceConfig.id_field_path),
+        nullable=True,
+        index=True,
+    )
+
+    experience_config_history_id = Column(
+        String,
+        ForeignKey(PrivacyExperienceConfigHistory.id_field_path),
+        nullable=True,
+        index=True,
+    )
+
+
+def get_privacy_notices_by_region_and_component(
+    db: Session, region: PrivacyNoticeRegion, component: ComponentType
+) -> Query:
+    """
+    Return relevant privacy notices that should be displayed for a certain
+    region in a given component type
+    """
+    return (
+        db.query(PrivacyNotice)
+        .filter(PrivacyNotice.regions.any(region.value))  # type: ignore[attr-defined]
+        .filter(
+            or_(
+                and_(
+                    component == ComponentType.overlay,
+                    PrivacyNotice.displayed_in_overlay,
+                ),
+                and_(
+                    component == ComponentType.privacy_center,
+                    PrivacyNotice.displayed_in_privacy_center,
+                ),
+            )
+        )
+    )
```

Comment:
This existing method extract to be reusued in multiple places.  Noting that PrivacyNotices and PrivacyExperiences still are not linked at all in the database.   We just filter an Experience at runtime to see which Notices share a region and a component at that time.